components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -94,9 +104,19 @@ public void index(Patient patient)
 
         for (Feature phenotype : patient.getFeatures()) {
             input.addField((phenotype.isPresent() ? "" : "negative_") + phenotype.getType(), phenotype.getId());
+            VocabularyTerm properTerm = this.ontologyService.getTerm(phenotype.getId());
+            if (properTerm != null) {
+                Set<VocabularyTerm> parents = properTerm.getAncestorsAndSelf();
+                for (VocabularyTerm term : parents) {
```

Comment:
This not quite correct. Terms should only be added to allAncestors if they are present. We don't want the full ancestry of the negative terms dominating the seeding. It might actually be best to put negative terms under a new solr field, `negative_phenotype_ancestors`.